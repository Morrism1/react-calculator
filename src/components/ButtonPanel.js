import Button from './Button';

export default function ButtonPanel() {
  return (
    <tbody className="inner-box">
      <tr>
        <Button name="AC" />
        <Button name="+/-" />
        <Button name="%" />
        <Button name="รท" className="blue-btn" />
      </tr>
      <tr>
        <Button name="7" />
        <Button name="8" />
        <Button name="9" />
        <Button name="X" className="blue-btn" />
      </tr>
      <tr>
        <Button name="4" />
        <Button name="5" />
        <Button name="6" />
        <Button name="-" className="blue-btn" />
      </tr>
      <tr>
        <Button name="1" />
        <Button name="2" />
        <Button name="3" />
        <Button name="+" className="blue-btn" />
      </tr>
      <tr>
        <Button name="0" />
        <Button name="." />
        <Button name="=" className="equal" />
      </tr>
    </tbody>
  );
}
