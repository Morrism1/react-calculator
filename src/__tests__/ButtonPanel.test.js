import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../components/ButtonPanel';
import '@testing-library/jest-dom/extend-expect';

const clickHandler = () => { ''; };

test('Render the ButtonPannel component  correctly', () => {
  const buttonpanel = renderer.create(<ButtonPanel handleCkick={clickHandler} />).toJSON();
  expect(buttonpanel).toMatchSnapshot();
});
