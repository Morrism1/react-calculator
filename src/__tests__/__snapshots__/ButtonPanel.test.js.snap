// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the ButtonPannel component  correctly 1`] = `
<tbody
  className="inner-box"
>
  <tr>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="blue-btn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </td>
  </tr>
  <tr>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="blue-btn"
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </td>
  </tr>
  <tr>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="blue-btn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </td>
  </tr>
  <tr>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="blue-btn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </td>
  </tr>
  <tr>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </td>
    <td
      colSpan={1}
    >
      <button
        className="light-mode"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </td>
    <td
      colSpan={2}
    >
      <button
        className="equal"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </td>
  </tr>
</tbody>
`;
